### DISCUSSION TITLE
Using BB as website / ecommerce admin

### DISCUSSION THREAD

**Discussion by smart-allocations-legacy** [Budibase/budibase#12538] [Help] [answered]
Hi,
I have a well designed ecommerce webiste, without any management, the website is a nodejs app speaking directly with postgress.
Adding the current database to BB is not an issue.
Today, the react app speaks with the nodejs, thoguh JWT token.

1. can my react app speak directly with the budibase api?  I didnt see one that allow to create records / update records.
2. if 1 is not possibe, I can continue writing from the nodejs to PG, but then I will loose BB capabilities such as hooks, security and more.
3.  I wonder how to handle the site users, should I create a table for them? how to handle the site roles and permissions? can I do it with the BB user management?

Is my usecase suitable with BB?

Thanks

**Comment by melohagan** [comment, marked as answer]
Hey @devtokenct 

1. You could use the public API: https://docs.budibase.com/reference/rowcreate

3. Users in Budibase are handled via [User management](https://docs.budibase.com/docs/user-management) in the portal. There is already a built-in *Users* table which cannot be edited. We also support Active Directory / SCIM: https://docs.budibase.com/docs/scim-provisioning

**Comment by smart-allocations-legacy** [comment]
@melohagan thanks
1. all the triggers / system flows work on this api like i was creating / updating from the dashboard.
2. Can I register external users / website users to this table via api? 
    if registring external users via api is poosible, what if I need more meta data on the user? can I create a related table with extra data about the user?
    
    Thanks again
   
   Thanks

**Reply by melohagan** [reply]
You can create users and assign them roles via the public API: https://docs.budibase.com/reference/usercreate
